nixos/tests/all-tests.nix

```
@@ -111,6 +111,7 @@ in {
   aesmd = runTestOn ["x86_64-linux"] ./aesmd.nix;
   agate = runTest ./web-servers/agate.nix;
   agda = handleTest ./agda.nix {};
+  agnos = handleTest ./agnos.nix {};
```

```suggestion
  agnos = discoverTests (import ./agnos.nix);
```
Since there is more than one test, this would be preferred.

You may need to check whether the use in passthru.tests still works